src/mon/OSDMonitor.cc

```diff
@@ -2137,6 +2152,37 @@ int OSDMonitor::get_osd_objectstore_type(int osd, string *type)
   return 0;
 }
 
+void OSDMonitor::get_filestore_list()
+{
+  for (unsigned osd = 0; osd < osdmap.get_num_osds(); ++osd) {
+    string objectstore_type;
+    int r = get_osd_objectstore_type(osd, &objectstore_type);
+    if (r == 0 && objectstore_type == "filestore") {
+      filestore_osds.insert(osd);
+    }
+  }
+}
+
+void OSDMonitor::check_for_filestore_osds(health_check_map_t *checks)
+{
+  ostringstream ss;
+  if (g_conf()->mon_health_warn_filestore_osds &&
+      filestore_osds.size() > 0) {
+    ss << filestore_osds.size() << " osds on filestore";
```

Comment:
Add information about forcing the use of 'wpq' scheduler for the osds on filestore.

Reply:
the warning should inform the user why this is a problem - it should mention that filestore is deprecated